merkle/tree_hasher.go

```diff
@@ -14,11 +14,24 @@
 
 package merkle
 
-// TreeHasher is the interface that the previous tree hasher struct implemented.
+// TreeHasher computes generic merkle trees.
 type TreeHasher interface {
-	HashEmpty() []byte
+	// EmptyRoot returns a special case for the root of an empty tree.
```

Comment:
Implementations don't necessarily need a special case. I think the word "allows" stuck in there might be clearer.